### DISCUSSION TITLE
How can I apply `allowedVersions` to some packages within a group?

### DISCUSSION THREAD

**Discussion by allanlewis** [renovatebot/renovate#24184] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'd like to have a group for Python type-checking dependencies - `mypy`, `types-*` and `*-stubs` - but control the version of Mypy. Specifically, I'd like to limit Mypy to `<1.5` but the types and stubs packages can use any version compatible with the other dependencies.

### What you tried so far.

I've made a Mypy group but I don't know how to include the types and stubs packages without also pinning them to the version of Mypy I want.

### Relevant debug logs

N/A

**Comment by viceice** [comment, marked as answer]
use two package rules

**Reply by allanlewis** [reply]
Sure, that's what I had assumed, but how do I do that while still grouping both Mypy and the types and stubs packages into one PR?

**Reply by rarkins** [reply]
First rule matches mypy and sets allowedVersions and groupName, second rule matches other packages and sets same groupName.

There's no "any version compatible with the other dependencies" rule

**Reply by allanlewis** [reply]
Ah so the key is using the same group name! Thanks 🙏

**Reply by rarkins** [reply]
Yes, although "grouping" sounds complicated, in some ways it's very simple: if the updates have the same branch name, then they'll be grouped in the same PR. The simplest way to get the same branch name is to specify groupName as otherwise the branch will be based on depName
